### ISSUE TITLE
cryptol-2.2.4 fails to build

### ISSUE THREAD

**Issue by mietek** [Homebrew/homebrew#47266]
This is an upstream issue. (https://github.com/GaloisInc/cryptol/issues/275)
```
src/Cryptol/Symbolic/Value.hs:85:9:
    Not in scope: ‘svReduceInPathCondition’

src/Cryptol/Symbolic/Value.hs:104:40:
    Not in scope: ‘svKind’
    Perhaps you meant ‘svAnd’ (imported from Data.SBV.Dynamic)
```

The 2.2.5 release fails to build, as well. (https://github.com/GaloisInc/cryptol/issues/292)
```
src/Cryptol/Symbolic.hs:115:37:
    Not in scope: ‘SBV.allSatWithAny’
    Perhaps you meant ‘SBV.satWithAny’ (imported from Data.SBV.Dynamic)
```

https://github.com/GaloisInc/cryptol/commit/1ffc086ed4b660919f7ef8113bfb48f9319fe908 builds, but doesn’t pass the tests. (https://github.com/GaloisInc/cryptol/issues/293)
```
Testing cryptol
==> Using the sandbox
==> /usr/local/Cellar/cryptol/2.2.5-1ffc086/bin/cryptol -b /private/tmp/cryptol20151222-62629-f13xkg/hello.icry
cryptol: user error (Missing response from solver)
Error: cryptol: failed
<0> expected but was
<1>.
```

**Comment by apjanke** [comment]
Is this the `cvc4` solver issue mentioned in #44503 (a cryptol 2.2.5 version bump pendinga resolution of that issue)?

**Comment by mietek** [comment]
I don’t know, because there is no output quoted, and the link to test results is 404, but it is likely to be the same issue.

**Comment by mietek** [comment]
Sorry, I didn’t see that issue before.  Some of the links are 404, but I found some [test output](https://travis-ci.org/Homebrew/homebrew/jobs/83192148#L932) in Jenkins, and it looks like a different issue.
```
Testing cryptol
==> Using the sandbox
/usr/bin/sandbox-exec -f /tmp/homebrew20151001-58096-inigdz.sb /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby -W0 -I /usr/local/Library/Homebrew -- /usr/local/Library/Homebrew/test.rb /usr/local/Library/Formula/cryptol.rb --verbose
==> /usr/local/Cellar/cryptol/2.2.5/bin/cryptol -b /private/tmp/cryptol20151001-58097-usfuxg/hello.icry
Error: cryptol: failed
Expected /Loading\ module\ Cryptol\nQ\.E\.D\.\nQ\.E\.D\.\n/ to match "Loading module Cryptol\n*** An error occurred.\n***  Failed to start the external solver: fd:6: hGetLine: end of file\n***  Make sure you can start \"/usr/local/bin/cvc4\"\n***  from the command line without issues.\n*** An error occurred.\n***  Failed to start the external solver: fd:9: hGetLine: end of file\n***  Make sure you can start \"/usr/local/bin/cvc4\"\n***  from the command line without issues.\n”.
```

**Comment by acfoltzer** [comment]
This is actually a separate issue from the `cvc4` problem, but since the `cvc4` problem is still outstanding, we can't get a Cryptol release past the buildbots. As a contingency, I'm working on switching over to Z3 for the default solver, which I recently had promoted to Homebrew.

**Comment by mietek** [comment]
Closed by [PR #47354](https://github.com/Homebrew/homebrew/pull/47354).
